providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/file_watcher.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+import os
+import threading
+import time
+import typing
+
+import yaml
+
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.event import ProviderEventDetails
+from openfeature.exception import ParseError, ProviderNotReadyError
+from openfeature.provider.provider import AbstractProvider
+
+from ..connector import FlagStateConnector
+from ..flags import FlagStore
+
+logger = logging.getLogger("openfeature.contrib")
+
+
+class FileWatcher(FlagStateConnector):
+    def __init__(
+        self,
+        file_path: str,
+        provider: AbstractProvider,
+        flag_store: FlagStore,
+        poll_interval_seconds: float = 1.0,
+    ):
+        self.file_path = file_path
+        self.provider = provider
+        self.poll_interval_seconds = poll_interval_seconds
+
+        self.last_modified = 0.0
+        self.flag_store = flag_store
+        self.emit_ready = False
+
+    def initialize(self, evaluation_context: EvaluationContext) -> None:
+        self.active = True
+        self.thread = threading.Thread(target=self.refresh_file, daemon=True)
+        self.thread.start()
+
```

Comment:
I would actually say starting regardless is intentional here? For example first read might fail or timeout, but state can still be repaired in background

Reply:
I think we shouldn't attempt to re-read the file if the initial read failed. Think about starting with an invalid file path. We should fail fast in this scenario and log the outcome so end-users know it failed. 

However, once we have the initialization, it's a different scenario. We should emit provider error and retry connection/read with back-offs.

Reply:
I love failing fast but imagine the file you read from is updated periodically by some other job and there is a syntax error. Now your pods couldn't scale out and new ones would all crash.

Perhaps giving the user the option to raise or swallow errors on initialization is a good idea? How do we handle this in other languages?

I'd be more worried about this behavior for something like grpc, http, k8s sources though, since those can be inherently unreliable

Reply:
Yeah that's a valid point. 

> How do we handle this in other languages?

Both Go & Java have fail fast strategy where if the initial read fails, provider initial fail, making evaluations to default. 

Don't take this as a blocker for this PR but this something to rethink @beeme1mr @toddbaert  any ideas ?

Reply:
I will mark this as closed for this PR based on the comments. @Kavindu-Dodan could you move the discussion somewhere else if you want to continue it outside the scope of this?